VALIDATION-SCHEMA.md

```diff
@@ -0,0 +1,116 @@
+is used to define the expected input and output conditions for a given custom training model.  This allows for easy communication of the expected input data from the creator to end users of a model.
+The specified conditions are also used at fit time to verify that the data provided meets the 
+constraints specified.  Inside DataRobot the requirements are used to verify that the blueprint inside the
+project provides the required datatypes, and conforms to the sparsity, number of columns and missing values constraints
+specified for the model.  
+
+## Validation Fields
+The full specification of all possible conditions and values for validation are listed in MODELING_METADATA.md.  Below is
+an explanation of how to utilize the various fields available. Only a single entry for any field may be present in the
+list for input requirements and for output requirements unless noted otherwise below.  
+
+### data_types
+The `data_types` field is used to specify those datatypes that are specifically expected, or those that
+are specifically disallowed.  Only a single entry is supported for this field in each specification.  
+The conditions used for data_types are:
+- EQUALS: A single data type is expected and must match the provided value
+- IN: All of the listed data types are expected in the dataframe.  An error will occur if one is missing or unexpected types are found.
+- NOT_EQUALS: The datatype for the input dataframe may not be this value.
+- NOT_IN: None of these datatypes are supported by the model.  
+
+### sparse
+The `sparse` field is used to define if the model supports input data that is of a sparse format, or if the
+model can create output data that is in a sparse format. For input the acceptable values are:
+A condition of EQUALS will always be used.  
+
+- FORBIDDEN: The model cannot handle a sparse matrix format, and will fail if one is provided.
+- SUPPORTED: The model can use a sparse matrix, or a dense matrix as input.
+- REQUIRED: This model only supports a sparse matrix as input and cannot use a dense dataframe.  
+
+For the output of the model the following values are allowed:
+- NEVER: This models output is always a dense dataframe
+- DYNAMIC: This model can output either a dense or sparse matrix.
+- ALWAYS: This model will never output a dense dataframe, and always a sparse matrix
+- IDENTITY: This model can output either a sparse or dense matrix, and the sparsity will match the input matrix.  If the 
+input is sparse this model will produce a sparse matrix, and if the input is dense it will output dense.  
+
+### number_of_columns
+The constraint for number of columns allows specifying if a specific number of columns are required or if the model has
+some maximum or minimum number of columns.  For time consuming models specifying a maximum number of columns can help
+keep performance reasonable.  Number of columns allows multiple entries to create ranges of values that are allwoed.  
```

Comment:
How would limiting of number of columns help time consuming models? It's not like a funnel or something, it would just fail if it gets more right? I don't understand

Reply:
If you know your model has training time dependent on dimensionality you might want to set an upper bound so that people you share it with don't use it on a dataset with too many columns.  It wouldn't necessarily fail outright.

Reply:
love it, this is smart